packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -334,8 +340,39 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
     this.authModelConfig.forEach((acm, modelName) => {
       const def = context.output.getObject(modelName)!;
       const modelHasSearchable = def.directives.some((dir) => dir.name.value === 'searchable');
+      const ownerFields = getOwnerFields(acm);
+      const filterInput = context.output.getInput(getFilterInputName(modelName));
+      if (filterInput) {
+        const updatedFilterInput = { ...filterInput, fields: [...filterInput.fields] };
+        ownerFields.forEach((ownerField) => {
+          if (!filterInput.fields.some((field) => field.name.value === ownerField)) {
+            updatedFilterInput.fields.push(makeInputValueDefinition(ownerField, makeNamedType('ModelStringInput')));
+          }
+        });
+        context.output.updateInput(updatedFilterInput);
+      }
+      const conditionInput = context.output.getInput(getConditionInputName(modelName));
+      if (conditionInput) {
+        const updatedConditionInput = { ...conditionInput, fields: [...conditionInput.fields] };
+        ownerFields.forEach((ownerField) => {
+          if (!conditionInput.fields.some((field) => field.name.value === ownerField)) {
+            updatedConditionInput.fields.push(makeInputValueDefinition(ownerField, makeNamedType('ModelStringInput')));
+          }
+        });
+        context.output.updateInput(updatedConditionInput);
+      }
+      const subscriptionFilterInput = context.output.getInput(getSubscriptionFilterInputName(modelName));
```

Comment:
Can you check the case where subscriptions are turned off? Do we still generate the subscription filter input type?